File: backend/src/fastCommandRouter.test.js
import test from "node:test";
import assert from "node:assert/strict";
import { parseFastCommand } from "./fastCommandRouter.js";

test("routes basic navigation without AI", () => {
  assert.deepEqual(parseFastCommand("go back"), { type: "BACK" });
  assert.deepEqual(parseFastCommand("go home"), { type: "HOME" });
  assert.deepEqual(parseFastCommand("open recent apps"), { type: "RECENTS" });
});

test("routes simple gestures and text", () => {
  assert.deepEqual(parseFastCommand("tap login"), { type: "TAP_TARGET", target: "login" });
  assert.deepEqual(parseFastCommand("double tap submit"), { type: "DOUBLE_TAP_TARGET", target: "submit" });
  assert.deepEqual(parseFastCommand("long press menu"), { type: "LONG_PRESS_TARGET", target: "menu" });
  assert.deepEqual(parseFastCommand("type hello world"), { type: "TYPE_TEXT", text: "hello world" });
  assert.deepEqual(parseFastCommand("wait 500 ms"), { type: "WAIT", waitMs: 500 });
});

test("does not steal complex natural-language tasks from the AI planner", () => {
  assert.equal(parseFastCommand("find the cheapest flight and book it"), null);
  assert.equal(parseFastCommand("what is 25 percent of 480"), null);
});

File: app/src/test/java/com/gamevision/companion/FastCommandRouterTest.kt
package com.gamevision.companion

import org.junit.Assert.assertEquals
import org.junit.Assert.assertNull
import org.junit.Test

class FastCommandRouterTest {
    @Test fun basicNavigationIsDeterministic() {
        assertEquals("BACK", FastCommandRouter.parse("go back")?.type)
        assertEquals("HOME", FastCommandRouter.parse("home")?.type)
        assertEquals("RECENTS", FastCommandRouter.parse("open recent apps")?.type)
        assertEquals("NOTIFICATIONS", FastCommandRouter.parse("notifications")?.type)
        assertEquals("QUICK_SETTINGS", FastCommandRouter.parse("open quick settings")?.type)
    }

    @Test fun typingAndWaitStayLocal() {
        val type = FastCommandRouter.parse("type hello world")
        assertEquals("TYPE_TEXT", type?.type)
        assertEquals("hello world", type?.text)
        assertEquals(500L, FastCommandRouter.parse("wait 500 ms")?.waitMs)
    }

    @Test fun complexRequestsRemainOnAiPath() {
        assertNull(FastCommandRouter.parse("find the cheapest flight and book it"))
        assertNull(FastCommandRouter.parse("what is 25 percent of 480"))
    }
}
